src/js/controllers/history.js

```diff
@@ -26,9 +28,30 @@ angular.module('copayApp.controllers').controller('historyController',
       return this.alternativeIsoCode;
     };
 
-    this.getTxHistory = function() {
+    this.checkCacheTxHistory = function() {
       var self = this;
+      storageService.getLastTransactions(fc.credentials.walletId, function(err, data) {
+        if (err) {
+          $log.debug('Error: ', err);
+          return;
+        }
+        self.txHistory = JSON.parse(data);
+      })
+    };
+
+    this.storeCacheTxHistory = function(txps) { 
+      storageService.storeLastTransactions(fc.credentials.walletId, txps, function(err) {
+        if (err) $log.debug('Error: ', err);
+      });
+    };
+
+    this.getTxHistory = function(firstTime) {
+      var self = this;
+      if (firstTime) {
+        self.checkCacheTxHistory();
+      }
       self.updatingTxHistory = true;
+      self.loadMore = false;
       $timeout(function() {
         fc.getTxHistory({
```

Comment:
I dont get why getTxHistory is called always, even if self.txHistory is set from the cache.